src/test/java/uk/gov/hmcts/reform/ucmc/service/PaymentsServiceTest.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.hmcts.reform.ucmc.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.payments.client.PaymentsClient;
+import uk.gov.hmcts.reform.payments.client.models.FeeDto;
+import uk.gov.hmcts.reform.payments.client.models.PaymentDto;
+import uk.gov.hmcts.reform.payments.client.request.CreditAccountPaymentRequest;
+import uk.gov.hmcts.reform.ucmc.config.PaymentsConfiguration;
+import uk.gov.hmcts.reform.ucmc.model.ClaimValue;
+import uk.gov.hmcts.reform.ucmc.request.RequestData;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class})
+class PaymentsServiceTest {
+    private static final String SERVICE = "service";
+    private static final String SITE_ID = "site_id";
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final FeeDto FEE_DATA = FeeDto.builder()
+        .version("1")
+        .code("CODE")
+        .calculatedAmount(BigDecimal.ONE)
+        .build();
+    private static final PaymentDto PAYMENT_DTO = PaymentDto.builder().reference("RC-1234-1234-1234-1234").build();
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private FeesService feesService;
+
+    @Mock
+    private PaymentsClient paymentsClient;
+
+    @Mock
+    private RequestData requestData;
+
+    @Mock
+    private PaymentsConfiguration paymentsConfiguration;
+
+    @InjectMocks
+    private PaymentsService paymentsService;
+
+    @BeforeEach
+    void setUp() {
+        given(feesService.getFeeDataByClaimValue(any())).willReturn(FEE_DATA);
+        given(paymentsClient.createCreditAccountPayment(any(), any())).willReturn(PAYMENT_DTO);
+        given(requestData.authorisation()).willReturn(AUTH_TOKEN);
+        given(paymentsConfiguration.getService()).willReturn(SERVICE);
+        given(paymentsConfiguration.getSiteId()).willReturn(SITE_ID);
+
+        paymentsService = new PaymentsService(
+            objectMapper,
+            feesService,
+            paymentsClient,
+            requestData,
+            paymentsConfiguration
+        );
```

Comment:
do you need this with inject mocks? If it's to do with resetting I seem to remember a resetMocks method

Reply:
Forgot to remove InjectMocks annotation. It needs to be done this way if you want to mix mocks with autowired beans. I had to autowire ObjectMapper, although I think we shouldn't do the mapping in the service, but rather pass mapped CaseData there.